src/core/types/object/json-object.d.ts

```diff
@@ -0,0 +1,14 @@
+export {};
+
+declare global {
+  // A type for Objects that can be JSON serialized or that come from
+  // JSON serialization. Requires the objects fields to be accessed with
+  // bracket notation object['name'] to make sure the fields do not get
+  // obfuscated.
+  // TODO(rcebulko): Closure Compiler used this and the @dict annotation to check
+  // that properties are only accessed using bracket notation. We need to
+  // verify this works with TypeScript/esbuild.
```

Comment:
We cannot enforce this guarantee unless we write a custom script via the new TS API.
That said, I don't think fields get dangerously obfuscated like they did with CC... so this is not really an issue.

cc @jridgewell